--- origsrc/java-gnome-4.0.20/configure	2011-07-10 20:13:12.000000000 -0500
+++ src/java-gnome-4.0.20/configure	2011-07-25 16:03:10.378252800 -0500
@@ -805,7 +805,7 @@ if ($os eq "gentoo") {
 			"gtkspell");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"libnotify");
 
@@ -853,7 +853,7 @@ if ($os eq "gentoo") {
 			"FIXME");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"FIXME");
 
@@ -917,7 +917,7 @@ if ($os eq "gentoo") {
 			"libgtksourceview2.0-dev");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"libnotify-dev");
 
@@ -968,7 +968,7 @@ if ($os eq "gentoo") {
 			"FIXME");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"FIXME");
 
@@ -1020,7 +1020,7 @@ if ($os eq "gentoo") {
 			"lib${cpu_arch}gtksourceview-2.0-devel");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"lib${cpu_arch}notify-devel");
 
@@ -1083,7 +1083,7 @@ if ($os eq "gentoo") {
 			"gtksourceview2-devel");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"libnotify-devel");
 
--- origsrc/java-gnome-4.0.20/src/bindings/org/gnome/notify/Notification.java	2011-07-10 20:13:12.000000000 -0500
+++ src/java-gnome-4.0.20/src/bindings/org/gnome/notify/Notification.java	2011-07-25 16:01:42.938251500 -0500
@@ -80,7 +80,7 @@ public class Notification extends org.gn
      */
 
     public Notification(String summary, String body, String icon) {
-        super(NotifyNotification.createNotification(summary, body, icon, null));
+        super(NotifyNotification.createNotification(summary, body, icon));
     }
 
     /**
@@ -103,7 +103,7 @@ public class Notification extends org.gn
      */
 
     public Notification(String summary, String body, String icon, Widget attach) {
-        super(NotifyNotification.createNotification(summary, body, icon, attach));
+        super(NotifyNotification.createNotification(summary, body, icon));
     }
 
     /**
@@ -118,7 +118,7 @@ public class Notification extends org.gn
      *             /Guidelines/MessageTray/Compatibility
      */
     public Notification(String summary, String body, String icon, StatusIcon statusIcon) {
-        super(NotifyNotification.createNotificationWithStatusIcon(summary, body, icon, statusIcon));
+        super(NotifyNotification.createNotification(summary, body, icon));
     }
 
     /**
@@ -144,7 +144,6 @@ public class Notification extends org.gn
      *             /Guidelines/MessageTray/Compatibility
      */
     public void attach(Widget attach) {
-        NotifyNotification.attachToWidget(this, attach);
     }
 
     /**
@@ -158,7 +157,6 @@ public class Notification extends org.gn
      *             /Guidelines/MessageTray/Compatibility
      */
     public void attach(StatusIcon statusIcon) {
-        NotifyNotification.attachToStatusIcon(this, statusIcon);
     }
 
     /**
@@ -168,7 +166,6 @@ public class Notification extends org.gn
      * @deprecated In GNOME 3
      */
     public void setGeometryHints(Screen screen, int x, int y) {
-        NotifyNotification.setGeometryHints(this, screen, x, y);
     }
 
     /**
--- origsrc/java-gnome-4.0.20/src/defs/NotifyNotification.defs	2011-07-10 20:13:12.000000000 -0500
+++ src/java-gnome-4.0.20/src/defs/NotifyNotification.defs	2011-07-25 16:02:18.802302800 -0500
@@ -15,19 +15,6 @@
     '("const-gchar*" "summary")
     '("const-gchar*" "body" (null-ok))
     '("const-gchar*" "icon" (null-ok))
-    '("GtkWidget*" "attach" (null-ok))
-  )
-)
-
-(define-function notify_notification_new_with_status_icon
-  (is-constructor-of "NotifyNotification")
-  (c-name "notify_notification_new_with_status_icon")
-  (return-type "NotifyNotification*")
-  (parameters
-    '("const-gchar*" "summary")
-    '("const-gchar*" "body" (null-ok))
-    '("const-gchar*" "icon" (null-ok))
-    '("GtkStatusIcon*" "status_icon")
   )
 )
 
@@ -42,35 +29,6 @@
   )
 )
 
-(define-method attach_to_widget
-  (of-object "NotifyNotification")
-  (c-name "notify_notification_attach_to_widget")
-  (return-type "none")
-  (parameters
-    '("GtkWidget*" "attach")
-  )
-)
-
-(define-method attach_to_status_icon
-  (of-object "NotifyNotification")
-  (c-name "notify_notification_attach_to_status_icon")
-  (return-type "none")
-  (parameters
-    '("GtkStatusIcon*" "status_icon")
-  )
-)
-
-(define-method set_geometry_hints
-  (of-object "NotifyNotification")
-  (c-name "notify_notification_set_geometry_hints")
-  (return-type "none")
-  (parameters
-    '("GdkScreen*" "screen")
-    '("gint" "x")
-    '("gint" "y")
-  )
-)
-
 (define-method show
   (of-object "NotifyNotification")
   (c-name "notify_notification_show")
