--- origsrc/java-gnome-4.0.13/Makefile	2009-08-26 22:19:47.000000000 -0500
+++ src/java-gnome-4.0.13/Makefile	2009-11-27 11:48:49.652468200 -0600
@@ -46,7 +46,7 @@ $(DESTDIR)$(PREFIX)/.java-gnome-install-
 
 install-java: build-java \
 	$(DESTDIR)$(JARDIR)/gtk-$(APIVERSION).jar \
-	$(DESTDIR)$(LIBDIR)/libgtkjni-$(VERSION).so
+	$(DESTDIR)$(LIBDIR)/cyggtkjni-$(VERSION).dll
 
 $(DESTDIR)$(JARDIR)/gtk-$(APIVERSION).jar: tmp/gtk-$(APIVERSION).jar
 	@/bin/echo -e "INSTALL\t$@"
@@ -56,7 +56,7 @@ $(DESTDIR)$(JARDIR)/gtk-$(APIVERSION).ja
 	@/bin/echo -e "SYMLINK\t$(@D)/gtk.jar -> gtk-$(APIVERSION).jar"
 	cd $(@D) && rm -f gtk.jar && ln -s gtk-$(APIVERSION).jar gtk.jar
 	
-$(DESTDIR)$(LIBDIR)/libgtkjni-$(VERSION).so: tmp/libgtkjni-$(VERSION).so
+$(DESTDIR)$(LIBDIR)/cyggtkjni-$(VERSION).dll: tmp/cyggtkjni-$(VERSION).dll
 	@/bin/echo -e "INSTALL\t$@"
 	cp -f $< $@
 
--- origsrc/java-gnome-4.0.13/build/faster	2009-08-26 22:19:47.000000000 -0500
+++ src/java-gnome-4.0.13/build/faster	2009-11-27 11:50:30.818644600 -0600
@@ -510,7 +510,7 @@ def compileBindingsObjects():
 
 	
 def linkSharedLibrary():
-	so = "tmp/libgtkjni-" + config['VERSION'] +  ".so"
+	so = "tmp/cyggtkjni-" + config['VERSION'] +  ".dll"
 
 	list = findFiles("tmp/objects/", ".o")
 	pairs = dependsListToSingleTarget(list, so)
--- origsrc/java-gnome-4.0.13/src/bindings/org/freedesktop/bindings/Plumbing.java	2009-08-26 22:19:48.000000000 -0500
+++ src/java-gnome-4.0.13/src/bindings/org/freedesktop/bindings/Plumbing.java	2009-11-27 11:53:46.302585400 -0600
@@ -152,7 +152,7 @@ public abstract class Plumbing
             libdir = new File(jar).getParent();
         }
 
-        library = new File(libdir, "libgtkjni-" + getVersion() + ".so");
+        library = new File(libdir, "cyggtkjni-" + getVersion() + ".dll");
         if (!library.exists()) {
             throw new UnsatisfiedLinkError("\n\n"
                     + "We anticipated loading the native portion of java-gnome from:\n" + library + "\n"
