From Gentoo Linux

=== modified file 'configure'
--- configure	2011-02-14 06:40:42 +0000
+++ configure	2011-02-14 22:27:29 +0000
@@ -805,7 +805,7 @@
 			"gtkspell");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"libnotify");
 
@@ -853,7 +853,7 @@
 			"FIXME");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"FIXME");
 
@@ -917,7 +917,7 @@
 			"libgtksourceview2.0-dev");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"libnotify-dev");
 
@@ -968,7 +968,7 @@
 			"FIXME");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"FIXME");
 
@@ -1020,7 +1020,7 @@
 			"lib${cpu_arch}gtksourceview-2.0-devel");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"lib${cpu_arch}notify-devel");
 
@@ -1083,7 +1083,7 @@
 			"FIXME");
 
 	check_system_library(@gnomedev_libs,
-			"libnotify >= 0.4.5 libnotify < 0.7.0",
+			"libnotify >= 0.7.0",
 			"LibNotify",
 			"FIXME");
 

=== modified file 'doc/examples/notify/ExampleLowBattery.java'
--- doc/examples/notify/ExampleLowBattery.java	2010-01-06 06:54:34 +0000
+++ doc/examples/notify/ExampleLowBattery.java	2011-03-28 16:05:00 +0000
@@ -61,7 +61,7 @@
          */
 
         notification = new Notification("Low Battery Example", "Your battery is low!",
-                "messagebox_warning", icon);
+                "messagebox_warning");
 
         /*
          * Quit the application after notification disappears.

=== modified file 'src/bindings/org/gnome/notify/Notification.java'
--- src/bindings/org/gnome/notify/Notification.java	2011-02-14 03:34:07 +0000
+++ src/bindings/org/gnome/notify/Notification.java	2011-03-13 22:23:48 +0000
@@ -80,41 +80,7 @@
      */
 
     public Notification(String summary, String body, String icon) {
-        super(NotifyNotification.createNotification(summary, body, icon, null));
-    }
-
-    /**
-     * Create a new notification.
-     * 
-     * <p>
-     * The summary appears on the titlebar of notification and body appears as
-     * its text. Icon may be a string defining a theme icon or the filename
-     * identifying the icon that appears next to text. Attach identifies the
-     * widget that the notification relates to.
-     * 
-     * <p>
-     * Note that all but summary can be <code>null</code>.
-     * 
-     * @since 4.0.12
-     * @deprecated In GNOME 3, notifications will be independent of widgets and status icons.
-     * See http://live.gnome.org/GnomeShell/Design/Guidelines/MessageTray/Compatibility
-     */
-
-    public Notification(String summary, String body, String icon, Widget attach) {
-        super(NotifyNotification.createNotification(summary, body, icon, attach));
-    }
-
-    /**
-     * Create a new notification attached to a {@link StatusIcon}. See
-     * {@link #Notification(String,String,String,Widget) Notification()} for
-     * other parameters.
-     * 
-     * @since 4.0.12
-     * @deprecated In GNOME 3, notifications will be independent of widgets and status icons.
-     * See http://live.gnome.org/GnomeShell/Design/Guidelines/MessageTray/Compatibility
-     */
-    public Notification(String summary, String body, String icon, StatusIcon statusIcon) {
-        super(NotifyNotification.createNotificationWithStatusIcon(summary, body, icon, statusIcon));
+        super(NotifyNotification.createNotification(summary, body, icon));
     }
 
     /**
@@ -131,39 +97,6 @@
     }
 
     /**
-     * Attaches Notification to a Widget setting hints to its location.
-     * 
-     * @since 4.0.12
-     * @deprecated In GNOME 3, notifications will be independent of widgets and status icons.
-     * See http://live.gnome.org/GnomeShell/Design/Guidelines/MessageTray/Compatibility
-     */
-    public void attach(Widget attach) {
-        NotifyNotification.attachToWidget(this, attach);
-    }
-
-    /**
-     * Attaches Notification to a {@link StatusIcon} setting hints to its
-     * location.
-     * 
-     * @since 4.0.12
-     * @deprecated In GNOME 3, notifications will be independent of widgets and status icons.
-     * See http://live.gnome.org/GnomeShell/Design/Guidelines/MessageTray/Compatibility
-     */
-    public void attach(StatusIcon statusIcon) {
-        NotifyNotification.attachToStatusIcon(this, statusIcon);
-    }
-
-    /**
-     * Sets the position of the notification to display on screen.
-     * 
-     * @since 4.0.12
-     * @deprecated In GNOME 3
-     */
-    public void setGeometryHints(Screen screen, int x, int y) {
-        NotifyNotification.setGeometryHints(this, screen, x, y);
-    }
-
-    /**
      * Display the notification on screen.
      * 
      * @since 4.0.12

=== modified file 'src/defs/NotifyNotification.defs'
--- src/defs/NotifyNotification.defs	2009-05-16 05:46:28 +0000
+++ src/defs/NotifyNotification.defs	2011-01-31 17:58:10 +0000
@@ -15,19 +15,6 @@
     '("const-gchar*" "summary")
     '("const-gchar*" "body" (null-ok))
     '("const-gchar*" "icon" (null-ok))
-    '("GtkWidget*" "attach" (null-ok))
-  )
-)
-
-(define-function notify_notification_new_with_status_icon
-  (is-constructor-of "NotifyNotification")
-  (c-name "notify_notification_new_with_status_icon")
-  (return-type "NotifyNotification*")
-  (parameters
-    '("const-gchar*" "summary")
-    '("const-gchar*" "body" (null-ok))
-    '("const-gchar*" "icon" (null-ok))
-    '("GtkStatusIcon*" "status_icon")
   )
 )
 
@@ -42,35 +29,6 @@
   )
 )
 
-(define-method attach_to_widget
-  (of-object "NotifyNotification")
-  (c-name "notify_notification_attach_to_widget")
-  (return-type "none")
-  (parameters
-    '("GtkWidget*" "attach")
-  )
-)
-
-(define-method attach_to_status_icon
-  (of-object "NotifyNotification")
-  (c-name "notify_notification_attach_to_status_icon")
-  (return-type "none")
-  (parameters
-    '("GtkStatusIcon*" "status_icon")
-  )
-)
-
-(define-method set_geometry_hints
-  (of-object "NotifyNotification")
-  (c-name "notify_notification_set_geometry_hints")
-  (return-type "none")
-  (parameters
-    '("GdkScreen*" "screen")
-    '("gint" "x")
-    '("gint" "y")
-  )
-)
-
 (define-method show
   (of-object "NotifyNotification")
   (c-name "notify_notification_show")
